server/src/main/java/io/crate/planner/WhereClauseOptimizer.java

```diff
@@ -137,6 +139,25 @@ public WhereClause toBoundWhereClause(DocTableInfo table,
                 );
             }
         }
+
+        public boolean hasPkSymbolsOnly() {
```

Comment:
imho, design wise, this has a signature of normal "getter" method, so it can be hidden from a consumer that there is complexity and "work" on every call. Personally I'd prefer this to be a static method called explicitly when needed, or at least calculate the flag once in the ctor and then have this one as a plain getter of the flag?

Reply:
Thanks for prompt response, I've turned it into a static methods for the calculation then in the end becomes a flag from `DetailedQuery`.